From 2a794a4e3f5f6a95f7bae0afb748ed7552d24f74 Mon Sep 17 00:00:00 2001
From: Shadi Ammouri <shadi@marvell.com>
Date: Mon, 28 Aug 2017 15:27:01 +0300
Subject: [PATCH 4/4] dts: agnic: ap806: Add dts entries for armada giu nic
 platform and uio

- The platform entry will be used by the agnic netdev driver.
- The uio entry will be used by the generic platform uio driver (to
expose agnic configuration registers to user-space).

Change-Id: Ifb6a231e7c4cfb4c2f8d604b7b0709b960151992
Signed-off-by: Shadi Ammouri <shadi@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/43483
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
---
 .../bindings/net/marvell-armada-giu-nic-uio.txt    |   21 ++++++++++++++++++++
 .../bindings/net/marvell-armada-giu-nic.txt        |   19 ++++++++++++++++++
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi      |   14 +++++++++++++
 3 files changed, 54 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/net/marvell-armada-giu-nic-uio.txt
 create mode 100644 Documentation/devicetree/bindings/net/marvell-armada-giu-nic.txt

diff --git a/Documentation/devicetree/bindings/net/marvell-armada-giu-nic-uio.txt b/Documentation/devicetree/bindings/net/marvell-armada-giu-nic-uio.txt
new file mode 100644
index 0000000..c22e708
--- /dev/null
+++ b/Documentation/devicetree/bindings/net/marvell-armada-giu-nic-uio.txt
@@ -0,0 +1,21 @@
+* Marvell Armada GIU NIC UIO
+
+This entry is used by the generic platform uio driver (uio_pdrv_genirq), in order to map the gnic configuration
+registers to user-space.
+
+required properties:
+=========================
+- compatible: should be "marvell,armada-giu-nic-uio"
+- reg: addresses and length of the register set for the device (must be the same as configured for
+  "marvell,armada-giu-nic" entry.
+  This is a set of two registers used by the management application to pass the base address of the configuration memory
+  space.
+
+Example:
+
+agnic-uio {
+	compatible = "marvell,armada-giu-nic-uio";
+	reg = <0x6F00A0 0x8>;
+	dma-coherent;
+	status = "okay";
+};
diff --git a/Documentation/devicetree/bindings/net/marvell-armada-giu-nic.txt b/Documentation/devicetree/bindings/net/marvell-armada-giu-nic.txt
new file mode 100644
index 0000000..f8dcd8d
--- /dev/null
+++ b/Documentation/devicetree/bindings/net/marvell-armada-giu-nic.txt
@@ -0,0 +1,19 @@
+* Marvell Armada GIU NIC (agnic)
+
+This entry is used by the loadable agnic driver (maintained out of the Linux tree)
+
+required properties:
+=========================
+- compatible: should be "marvell,armada-giu-nic"
+- reg: addresses and length of the register set for the device.
+  This is a set of two registers used by the management application to pass the base address of the configuration memory
+  space.
+
+Example:
+
+agnic-plat {
+	compatible = "marvell,armada-giu-nic";
+	reg = <0x6F00A0 0x8>;
+	dma-coherent;
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index c0d7cff..7402a72 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -489,6 +489,20 @@
 				reg = <0x6f8200 0x8>;
 				status = "okay";
 			};
+
+			agnic-plat {
+				compatible = "marvell,armada-giu-nic";
+				reg = <0x6F00A0 0x8>;
+				dma-coherent;
+				status = "okay";
+			};
+
+			uio_agnic_0 {
+				compatible = "marvell,armada-giu-nic-uio";
+				reg = <0x6F0000 0x1000>;
+				reg-names = "agnic_regs";
+				status = "okay";
+			};
 		};
 	};
 };
-- 
1.7.9.5

